sematic/db/models/edge.py

```diff
@@ -19,20 +19,20 @@ class Edge(Base, JSONEncodableMixin):
 
     # Edge endpoints
     source_run_id: Optional[str] = Column(
-        types.String(), ForeignKey("artifacts.id"), nullable=True
+        types.String(), ForeignKey("runs.id"), nullable=True
```

Comment:
see comment about these changes in the migration.  this was a typo/bug that's now fixed.

Reply:
we found a few instances where the source/dest run id didn't actually exist, but these were quite few in number.  for any such instances, we had to run queries to resolve the discrepancy.  however, there is no risk of the source/dest run ids actually pointing at an artifact instead of a run, because all these ids are uuids, which means that it's a statistical impossibility that we have a run and an artifact in the db with the same id.